jobclient/python/__init__.py

```diff
@@ -0,0 +1,206 @@
+
+        Parameters
+        ----------
+        :param host: The hostname of the Cook instance.
+        :type host: str
+        :param port: The port at which the Cook instance is listening.
+        :type port: int
+        :param job_endpoint: The endpoint to be reached when scheduling new
+            jobs or getting the status of running jobs. Defaults to `/jobs`.
+        :type job_endpoint: str, optional
+        :param delete_endpoint: The endpoint to be reached when terminating
+            running jobs. Defaults to `/rawscheduler`
+        """
+        self.__netloc = f'{host}:{port}'
+        self.__job_endpoint = job_endpoint
+        self.__delete_endpoint = delete_endpoint
+
+    def submit(self, *,
+               command: str,
+               cpus: float,
+               mem: float,
+               max_retries: int,
+
+               uuid: Optional[UUID] = None,
+               env: Optional[Dict[str, str]] = None,
+               labels: Optional[Dict[str, str]] = None,
+               max_runtime: Optional[int] = None,
+               name: Optional[str] = None,
+               priority: Optional[int] = None,
+               application: Optional[Application] = None) -> UUID:
+        """Submit a single job to Cook.
+
+        Required Parameters
+        -------------------
+        :param command: The command to run on Cook.
+        :type command: str
+        :param cpus: The number of CPUs to request from Cook.
+        :type cpus: float
+        :param mem: The amount of memory, in GB, to request from Cook.
+        :type mem: float
+        :param max_retries: The number of times this job should be retried
+            before failing.
+        :type max_retries: int
+        Optional Parameters
+        -------------------
+        :param uuid: The UUID of the job to submit. If this value is not
+            provided, then a random UUID will be generated.
+        :type uuid: UUID, optional
+        :param env: Environment variables to set within the job's context,
+            defaults to None.
+        :type env: Dict[str, str], optional
+        :param labels: Labels to assign to the job, defaults to None.
+        :type labels: Dict[str, str], optional
+        :param max_runtime: The maximum number seconds this job should be
+            allowed to run, defaults to None.
+        :type max_runtime: int, optional
+        :param name: A name to assign to the job, defaults to None.
+        :type name: str, optional
+        :param priority: A priority to assign to the job, defaults to None.
+        :type priority: int, optional
+        :param application: Application information to assign to the job,
+            defaults to None.
+        :type application: Application, optional
+        Output
+        ------
+        :return: The UUID of the newly-created job.
+        :rtype: UUID
+        """
+        uuid = str(uuid if uuid is not None else util.make_temporal_uuid())
```

Comment:
👍

Reply:
Could this be simplified to:

```suggestion
        uuid = str(uuid or util.make_temporal_uuid())
```

Reply:
:+1: